specs/WebResourceRequested-CustomScheme.md

```diff
@@ -0,0 +1,322 @@
+  // Represents the registration of a custom scheme with the CoreWebView2Environment.
+  // This allows the WebView2 app to be able to handle
+  // WebResourceRequested event for requests with the specified scheme and
+  // be able to navigate the WebView2 to the custom scheme. Once the environment
+  // is created, the registrations are valid and immutable throughout the
+  // lifetime of the associated WebView2s' browser process and any WebView2
+  // environments sharing the browser process must be created with identical
+  // custom scheme registrations, otherwise the environment creation will fail.
+  // If there are multiple entries for the same scheme in the registrations list, the environment creation will also fail.
+  // The URIs of registered custom schemes will be treated similar to http URIs for their origins.
+  // They will have tuple origins for URIs with host and opaque origins for
+  // URIs without host as specified in [7.5 Origin - HTML Living Standard](https://html.spec.whatwg.org/multipage/origin.html)
+  // Example:
+  // custom-scheme-with-host://hostname/path/to/resource has origin of custom-scheme-with-host://hostname
+  // custom-scheme-without-host:path/to/resource has origin of custom-scheme-without-host:path/to/resource
+  // For WebResourceRequested event, the request URIs and filter URIs with custom schemes
+  // will be normalized according to RFC3986.
+
+  // The name of the custom scheme to register.
+  [propget] HRESULT SchemeName([out, retval] LPCWSTR* schemeName);
+  [propput] HRESULT SchemeName([in] LPCWSTR value);
+
+  // Whether the scheme will be treated as a secure context like https.
+  [propget] HRESULT IsSecure([out, retval] BOOL* isSecure);
+  // Set if the scheme will be treated as a secure context.
+  [propput] HRESULT IsSecure([in] BOOL value);
+
+  // If `isCorsEnabled` is set, the scheme will be allowed for requests subject to
+  // CORS rules as specified in [Cross-Origin Resource Sharing - MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS).
+  // If it is not set, CORS requests with this scheme will be blocked.
+  [propget] HRESULT IsCorsEnabled([out, retval] BOOL* isCorsEnabled);
+  // Set CORS enabled for the scheme.
+  [propput] HRESULT IsCorsEnabled([in] BOOL value);
+
+  // List of origins that are allowed to issue requests with the custom scheme.
+  // Except origins with custom scheme itself and no-origin requests,
+  // the origin of any request to the custom scheme URL
+  // needs to be in this list. Note that cross-origin restrictions still apply. 
+  // Origins are specified as a string in the format of scheme://host:port.
+  // The origins are string pattern matched with `*` (matches 0 or more characters) and `?` (matches 0 or 1 character) wildcards.
+  // For example, "http://*.example.com:80".
+  // Any origin from the custom scheme itself is allowed by default. If the
+  // array is empty, no other origin except origins from same scheme are allowed.
```

Comment:
Should say here that the caller owns the memory of allowedOrigins and that they should deallocate with CoTaskMemFree. Is this one block of memory containing the strings and the pointers? Or does the caller need to deallocate each string in the array and then deallocate the array?